### PR TITLE
clippy: Fix warnings in `components/script/dom/request.rs`

### PR DESCRIPTION
**Pull request by Aaryakhandelwal**
Major changes are: 
1)  changed the implementation of Into<_> to From statement as the former is included in the latter by default.
2) Dereferenced wherever necessary.

---

- [X] `./mach build -d` does not report any errors
- [X] `./mach test-tidy` does not report any errors
- [X] These changes are part of #31500.

- [X] There are tests for these changes OR
- [X] These changes do not require tests because they only fix clippy warnings.

### CODE REVIEW

File: components/script/dom/request.rs [hunk 1]
```diff
@@ -666,9 +666,9 @@ impl BodyMixin for Request {
     }
 }
 
-impl Into<NetTraitsRequestCache> for RequestCache {
-    fn into(self) -> NetTraitsRequestCache {
-        match self {
+impl From<RequestCache> for NetTraitsRequestCache {
+    fn from(val: RequestCache) -> Self {
+        match val {
```
Comment (mrobinson) [inline components/script/dom/request.rs:671]:
```suggestion
impl From<RequestCache> for NetTraitsRequestCache {
    fn from(value: RequestCache) -> Self {
        match value {
```

**Review by mrobinson — COMMENTED** [review]
Looks great! I only have one nit request for these changes. Let's avoid abbreviating in this case since "value" is only two characters longer and increases clarity.

**Review by mrobinson — CHANGES_REQUESTED** [review]
Apologies. I didn't set the proper review flag before.

File: components/script/dom/request.rs (hunk 1 above)
Comment (mrobinson) [reply components/script/dom/request.rs]: I've gone ahead and made these changes.

**Review by mrobinson — APPROVED** [review]

File: components/script/dom/request.rs (hunk 1 above)
Comment (Aaryakhandelwal) [reply components/script/dom/request.rs]: Apologies, I was preoccupied with some personal work and couldn't attend to the changes requested. 
Thank you for making the commit.

Comment (mrobinson) [reply components/script/dom/request.rs]: No worries. Thank you for the contribution!

**Pull request merged**